### PR TITLE
refactor: extract save function from component ManageTokensModal

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

Since we would like to use `ManageTokensModal` in a more generic way, we first extract the save function and use it as service.

# Changes

- Mirror function `save` to `saveIcrcCustomToken `.
- Adapt component `ManageTokensModal`.

# Tests

Manual test on replica showed unchanged behavior.

### CODE REVIEW

File: src/frontend/src/icp/services/ic-custom-tokens.services.ts [hunk 1]
```diff
@@ -41,3 +46,53 @@ export const saveCustomTokens = async ({
...
+	onError
+}: {
+	tokens: Pick<IcrcCustomToken, 'enabled' | 'version' | 'ledgerCanisterId' | 'indexCanisterId'>[];
+	updateSaveProgressStep: (step: ProgressStepsAddToken) => void;
+	modalNext: () => void;
+	onSuccess: () => void;
+	onError: () => void;
+}) => {
+	const $i18n = get(i18n);
+	const $authStore = get(authStore);
```
Comment (peterpeterparker) [inline src/frontend/src/icp/services/ic-custom-tokens.services.ts:64]: can you pass the `identity` as parameter of the function for consistency reason?

**Review by peterpeterparker — APPROVED** [review]
LGTM, minor suggestion

File: src/frontend/src/icp/services/ic-custom-tokens.services.ts (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp/services/ic-custom-tokens.services.ts:64]: doing it, and do you want to keep the same name for the `updateSaveProgressStep` and call it `progress`?

Comment (peterpeterparker) [reply src/frontend/src/icp/services/ic-custom-tokens.services.ts:64]: Ah yeah why not, it's generally what I used so far

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp/services/ic-custom-tokens.services.ts:64]: cool

**Pull request merged**